docs/side_quests/nf-test.md

```diff
@@ -0,0 +1,1074 @@
+
+In this part of the training, we're going to show you how to use nf-test to write module-level tests for the three processes in our pipeline.
+
+---
+
+## 0. Warmup
+
+Let's move into the project directory.
+
+```bash
+cd side-quests/nf-test
+```
+
+The `nf-test` directory has the file content like:
+
+```console title="Directory contents"
+nf-test
+├── greetings.csv
+├── main.nf
+└── nextflow.config
+```
+
+For a detailed description of the files, see the [warmup from Hello Nextflow](../hello_nextflow/00_orientation.md).
+
+### 0.1. Run the workflow
+
+Let's run the workflow to make sure it's working as expected.
+
+```bash
+nextflow run main.nf
+```
+
+```console title="Result of running the workflow"
+ N E X T F L O W   ~  version 24.10.2
+
+Launching `main.nf` [soggy_linnaeus] DSL2 - revision: bbf79d5c31
+
+executor >  local (6)
+[f7/c3be66] sayHello (3)       | 3 of 3 ✔
+[cd/e15303] convertToUpper (3) | 3 of 3 ✔
+```
+
+CONGRATULATIONS! You just ran a test!
+
+"Wait, what? I just ran the workflow and it worked! How is that a test?"
+
+Good question!
+
+Let's break down what just happened.
+
+You ran the workflow with the default parameters, you confirmed it worked and you're happy with the results. This is the essence of testing. If you worked through the Hello Nextflow training course, you'll have noticed we always started every section by running the workflow we were using as a starting point, to confirm everything is set up correctly.
+
+Testing software essentially does this process for us. Let's replace our simple `nextflow run main.nf` with a standardised test provided by nf-test.
+
+### Takeaway
+
+You should be able to 'test' a pipeline by manually running it.
+
+### What's next?
+
+Initialize `nf-test`.
+
+---
+
+## 1.0. Initialize `nf-test`
+
+The `nf-test` package provides an initialization command that sets up a few things in order for us to start developing tests for our project.
+
+```bash
+nf-test init
+```
+
+This should produce the following output:
+
+```bash
+🚀 nf-test 0.9.2
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+Project configured. Configuration is stored in nf-test.config
+```
+
+It also creates a `tests` directory containing a configuration file stub.
+
+## 1.1. Generate an nf-test
+
+`nf-test` comes with a set of tools for building nf-test files, saving us the majority of the work. These come under the subcommand `generate`. Let's generate a test for the pipeline:
+
+```bash
+nf-test generate pipeline main.nf
+```
+
+```console title="Output"
+> nf-test generate pipeline main.nf
+
+Load source file '/workspaces/training/side-quests/nf-test/main.nf'
+Wrote pipeline test file '/workspaces/training/side-quests/nf-test/tests/main.nf.test
+
+SUCCESS: Generated 1 test files.
+```
+
+This will create a `tests` directory with a `main.nf.test` and a `nextflow.config` file. This is our pipeline level test file. If you run `tree tests/` you should see something like this:
```

Comment:
The `tests` directory has already been created by `nf-test init`, no? That's what is said above. I also tried it here and saw this.

```suggestion
This will create a `main.nf.test` file within the `tests` directory. This is our pipeline level test file. If you run `tree tests/` you should see something like this:
```

Reply:
Yes, you're correct- thanks!